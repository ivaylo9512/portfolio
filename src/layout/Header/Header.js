import { changeSlide, getCurrentSlide, isProjectsSection, resetSlides, toggleSections } from 'app/slices/animationsSlice';
import { useDispatch, useSelector } from 'react-redux';
import Arrow from '@svgIcons/Arrow';
import Circle from '@svgIcons/Circle';
import Rect from '@svgIcons/Rect';
import './Header.scss';

const Header = () => {
    const dispatch = useDispatch(); 
    const isProjects = useSelector(isProjectsSection);
    const currentSlide = useSelector(getCurrentSlide);

    const onToggleSections = () => {
        window.scrollTo(0, 0);
        dispatch(toggleSections());

        if(isProjectsSection){
            dispatch(resetSlides())
        }
    }

    const onChangeSlide = (byAmount) => {
        dispatch(changeSlide(byAmount))
    }

    return(
        <header className={`${currentSlide}-slide`}>
            <nav>
                <ul>
                    <ul className='left-menu'> 
                         <li>
                            <a href='/#' data-testid='sections-toggle' className='sections-toggle' onClick={onToggleSections}>
                                <Rect width={200} height={90} radius={40} />
                                <span data-testid={isProjects ? 'cv' : 'projects'}>{isProjects ? 'Cv' : 'Projects'}
                                    <Arrow />
                                </span>
                            </a>
                        </li>
                        <li className="slide-left">
                            <button onClick={() => onChangeSlide(-1)}>
                                <Circle />
                                <Arrow />
                            </button>
                        </li>

                    </ul>
                    <li className='slide-right'>
                        <button onClick={() => onChangeSlide(1)}>
                            <Circle />
                            <Arrow />
                        </button>
                    </li>
                </ul>
            </nav>
        </header>
    )
}

export default Header