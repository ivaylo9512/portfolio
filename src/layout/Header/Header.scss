header {
    position: fixed;
    transition: 1.5s all;
    width: 100%;
    z-index: 3;
    padding: env(safe-area-inset-top) env(safe-area-inset-right) env(safe-area-inset-bottom) calc(env(safe-area-inset-left) + 1%);
    box-sizing: border-box;
    padding-top: 1vh;

    button, a{
        height: 6vh;
        
        &:hover,
        &:focus{
            transform: scale(1.1);
        }
    }

    a{
        display: block;
        position: relative;
        align-items: center;
        justify-content: center;

        span{
            color: white;
            border-radius: 1vw;
            align-items: center;
            font-size: 1.8vh;
            font-weight: bold;
            display: flex;
            justify-content: center;
            position: absolute;
            width: 100%;
            left: 0;
            z-index: -1;
            top: 0;
            height: 100%;
        }

        .rect{
            position: relative;
            height: 100%;
            z-index: -1;
       
            rect{
                filter: url(#project-shadow);
                transition: 1.5s all;
            }
        }

        .arrow{
            height: 42%;
            fill: white;
            transform: rotate(90deg);
            margin-left: 0.3vw;
            transition: transform 1.5s;
        }
    }

    button{
        position: relative;
        display: block;

        .circle{
            height: 100%;
        }    
        
        .arrow{
            position: absolute;
            fill: white;
            width: 31%;
            top: 50%;
            left: 50%;
            transform: translate(-50%, -50%) rotate(90deg);
        }
    }
    
    ul{
        display: flex;
    }
    
    .left-menu{
        transition: transform 1.5s;
    }

    .slide-left,
    .slide-right{
        opacity: 0;
    }

    .slide-right{
        margin-left: auto;
        margin-right: 1%;
        fill: var(--clr-profile-primary);
    }
    
    .slide-left{
        margin-left: 1vh;
        fill: var(--clr-tasks-primary);
        transform:  rotate(-180deg);
        transition: opacity 0s;
    }

    .left-menu{
        transition: transform 1.5s;
        transform: translateX(calc(97vw - 70% - env(safe-area-inset-right) - env(safe-area-inset-left)));
    }

    .slide-right,
    .slide-left{
        opacity: 0;
        pointer-events: none;

        circle{
            filter: url(#profile-shadow);
        }
    }

    @media only screen and (max-width : 896px) and (orientation : landscape) { 
        button,
        a{
            height: 10vh;
        }
    }
}
.projects-section header {
    .left-menu{
        transform: translateX(0);
    }

    .slide-left{
        transition: opacity 0.5s;
    }

    .sections-toggle{
        rect{
            fill: var(--clr-tasks-primary);
        }

        .arrow{
            transform: rotate(-90deg);
        }
    }
}

@media only screen and (min-aspect-ratio: 10 / 6 ) {
    .projects-section header {
        .slide-right {
            opacity: 1;
            transform: translateX(0);
            pointer-events: all;
            transition: opacity 0.5s 1s;
        }

        .left-menu{
            transform: translateX(0);
        }

        .sections-toggle{        
            rect{
                fill: var(--clr-tasks-primary);
            }
        
            .arrow{
                transform: rotate(-90deg);
            }
        }

        &.second-slide {
            .slide-right {
                circle{
                    transition: fill 0.5s 0.7s;
                    fill: var(--clr-extensions-primary);
                }
            }
        
            .sections-toggle{
                rect{
                    fill: var(--clr-extensions-primary);
                }
            }
        
            .slide-left{
                opacity: 1;
                pointer-events: all;
        
                circle{
                    fill: var(--clr-extensions-primary);
                }
            }
        
            .left-menu{    
                transform: translateX(calc(98vw - 100% - 7vh - env(safe-area-inset-left) - env(safe-area-inset-right)));
            }
        }
        
        &.third-slide {
            .sections-toggle{
                rect{
                    fill: var(--clr-extensions-primary);
                }
            }
        
            .slide-right {
                pointer-events: none;
                opacity: 0;
                transition: opacity 0.5s;
            }
        
            .slide-left{
                opacity: 1;
                pointer-events: all;
                
                circle{
                    fill: var(--clr-extensions-primary);
                }
            }
        }
    }
}

@media only screen and (max-width : 896px) and (orientation : landscape) { 
    header{
        button,
        a{
            height: 10vh;
        }
    }

    .projects-section header {
        &.second-slide {
            .left-menu{    
                transform: translateX(calc(98vw - 100% - 12vh - env(safe-area-inset-left) - env(safe-area-inset-right)));
            }
        }
    }
}