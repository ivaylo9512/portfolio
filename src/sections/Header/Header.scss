header {
    position: fixed;
    width: 100%;
    z-index: 3;

    ul{
        display: flex;
    }

    ul > :last-child{
        margin-left: auto;
    }

    a{
        padding: 5vh;
        background-color: black;
    }
}