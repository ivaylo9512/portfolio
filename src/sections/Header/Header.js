import { isProjectsSection, toggleSections } from 'app/slices/animationsSlice';
import { useDispatch, useSelector } from 'react-redux';
import Arrow from 'svgIcons/Arrow';
import './Header.scss';

const Header = () => {
    const dispatch = useDispatch(); 
    const isProjects = useSelector(isProjectsSection);

    const onToggleSections = () => {
        dispatch(toggleSections());
    }

    return(
        <header>
            <nav>
                <ul>
                    <li>
                        <a>
                            <Arrow />
                        </a>
                    </li>
                    <li>
                        <a data-testid='sections-toggle' onClick={onToggleSections}>
                            {isProjects 
                                ? <span data-testid='cv'>Cv'</span>
                                : <span data-testid='projects'>'Projects'</span>
                            }
                        </a>
                    </li>
                    <li>
                        <a>
                            <Arrow />
                        </a>
                    </li>
                </ul>
            </nav>
        </header>
    )
}

export default Header