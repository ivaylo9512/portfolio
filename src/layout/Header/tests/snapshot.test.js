import { shallow } from 'enzyme';
import Header from '../Header';
import * as Redux from 'react-redux';

describe('Header snapshot tests', () => {
    let selectorSpy;
   
    beforeEach(() => {
        jest.spyOn(Redux, 'useDispatch').mockReturnValue(jest.fn());
        selectorSpy = jest.spyOn(Redux, 'useSelector');
    })

    const createWrapper = (state) => {
        selectorSpy.mockReturnValue(state);
        
        return shallow(
            <Header />
        )
    }

    it('should match snapshot with projects secton', () => {
        const wrapper = createWrapper(true);

        expect(wrapper).toMatchSnapshot();
    })

    it('should match snapshot with cv secton', () => {
        const wrapper = createWrapper(false);

        expect(wrapper).toMatchSnapshot();
    })
})