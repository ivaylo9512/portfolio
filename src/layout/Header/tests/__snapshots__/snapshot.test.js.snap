// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot tests should match snapshot with cv secton 1`] = `
<header
  className="false-slide"
>
  <nav>
    <ul>
      <ul
        className="left-menu"
      >
        <li>
          <a
            className="sections-toggle"
            data-testid="sections-toggle"
            href="/#"
            onClick={[Function]}
          >
            <Rect
              height={90}
              radius={40}
              width={200}
            />
            <span
              data-testid="projects"
            >
              Projects
              <Arrow />
            </span>
          </a>
        </li>
        <li
          className="slide-left"
        >
          <button
            onClick={[Function]}
          >
            <Circle />
            <Arrow />
          </button>
        </li>
      </ul>
      <li
        className="slide-right"
      >
        <button
          onClick={[Function]}
        >
          <Circle />
          <Arrow />
        </button>
      </li>
    </ul>
  </nav>
</header>
`;

exports[`Header snapshot tests should match snapshot with projects secton 1`] = `
<header
  className="true-slide"
>
  <nav>
    <ul>
      <ul
        className="left-menu"
      >
        <li>
          <a
            className="sections-toggle"
            data-testid="sections-toggle"
            href="/#"
            onClick={[Function]}
          >
            <Rect
              height={90}
              radius={40}
              width={200}
            />
            <span
              data-testid="cv"
            >
              Cv
              <Arrow />
            </span>
          </a>
        </li>
        <li
          className="slide-left"
        >
          <button
            onClick={[Function]}
          >
            <Circle />
            <Arrow />
          </button>
        </li>
      </ul>
      <li
        className="slide-right"
      >
        <button
          onClick={[Function]}
        >
          <Circle />
          <Arrow />
        </button>
      </li>
    </ul>
  </nav>
</header>
`;
